File: common/src/main/java/app/qwertz/quickjoin/fallback/JsonFallback.java
package app.qwertz.quickjoin.fallback;

public class JsonFallback {

    public static String getJsonFallback() {
        String jsonFallback = getResourceAsText("/assets/quickjoin/guis.json");
        return jsonFallback != null ? jsonFallback : "{}";
    }

    private static String getResourceAsText(String path) {
        try {
            java.io.InputStream is = JsonFallback.class.getResourceAsStream(path);
            if (is == null) return null;
            java.util.Scanner s = new java.util.Scanner(is).useDelimiter("\\A");
            return s.hasNext() ? s.next() : null;
        } catch (Exception e) {
            return null;
        }
    }
}

File: src/main/kotlin/app/qwertz/quickjoin/fallback/JsonFallBack.kt
package app.qwertz.quickjoin.fallback


fun getJsonFallback(): String {
    val jsonfallback = getResourceAsText("/assets/quickjoin/guis.json") ?: "{}"
    return jsonfallback
}

private fun getResourceAsText(path: String): String? =
        object {}.javaClass.getResource(path)?.readText()
